### PR TITLE
Optimize specifing port range with binary mask

### PR DESCRIPTION
**Pull request by hengyoush**
for issue https://github.com/chaosblade-io/chaosblade/issues/885
主要改动如下:
1. 增加buildMaskForRange函数, 计算表达给定整数区间的掩码.
2. 端口等参数采用区间列表的形式, 方便计算掩码
3. 增加TestBuildMaskForRange测试, 用于验证buildMaskForRange函数的正确性

### CODE REVIEW

File: exec/network/tc/network_tc.go [hunk 1]
```diff
@@ -188,15 +184,14 @@ func buildExcludeFilterToNewBand(netInterface string, excludePorts []string, exc
...
-			tc filter add dev %s parent 1: prio 4 protocol ip u32 match ip sport %s 0xffff flowid 1:4`,
-			args, netInterface, port, netInterface, port)
+	for _, portRange := range excludePortRanges {
+		masks := buildMaskForRange(portRange[0], portRange[1])
+		for _, mask := range masks {
+		  args = fmt.Sprintf(
+			  `%s && \
+			  tc filter add dev %s parent 1: prio 4 protocol ip u32 match ip dport %s 0xffff flowid 1:4 && \
+			  tc filter add dev %s parent 1: prio 4 protocol ip u32 match ip sport %s 0xffff flowid 1:4`,
+			  args, netInterface, port, netInterface, port)
```
Comment (MandssS) [inline exec/network/tc/network_tc.go:194]: 这里port从哪来的？

File: exec/network/tc/network_tc.go [hunk 2]
```diff
@@ -188,15 +184,14 @@ func buildExcludeFilterToNewBand(netInterface string, excludePorts []string, exc
...
 	}
 	return args
 }
```
Comment (MandssS) [inline exec/network/tc/network_tc.go:197]: 少了一个 }

File: exec/network/tc/network_tc.go (hunk 1 above)
Comment (hengyoush) [reply exec/network/tc/network_tc.go]: done
和其他分支的代码冲突了

File: exec/network/tc/network_tc.go (hunk 2 above)
Comment (hengyoush) [reply exec/network/tc/network_tc.go:198]: done

**Review by binbin0325 — COMMENTED** [review]
ip has the same problem (too long parameters), is there a solution similar to port?

**Comment by hengyoush** [comment]
> ip has the same problem (too long parameters), is there a solution similar to port?
For ip we can use  subnet mask?

**Review by MandssS — APPROVED** [review]

**Pull request merged**